### PR TITLE
PP-12403 Add pkl module for pack broker

### PR DESCRIPTION
**Pull request by kbottla**
## WHAT
- Added pkl module for pact broker

fly -t pay-deploy set-pipeline -c ./pact-broker.yml -p pact-broker --dry-run

```diff
resources:
  resource pact-broker-pipeline has been removed:
- icon: github
- name: pact-broker-pipeline
- source:
-   branch: master
-   paths:
-   - ci/pipelines/pact-broker.yml
-   uri: https://github.com/alphagov/pay-ci
- type: git

  resource pact-broker-src has changed:
  icon: github
  name: pact-broker-src
  source:
    branch: master
+   password: ((github-access-token))
    paths:
    - ci/docker/pact-broker/*
    uri: https://github.com/alphagov/pay-ci
+   username: alphagov-pay-ci-concourse
  type: git

  resource pay-ci has changed:
  icon: github
  name: pay-ci
  source:
    branch: master
+   password: ((github-access-token))
    uri: https://github.com/alphagov/pay-ci
+   username: alphagov-pay-ci-concourse
  type: git

  resource pipeline-source has been added:
+ icon: github
+ name: pipeline-source
+ source:
+   branch: master
+   password: ((github-access-token))
+   paths:
+   - ci/pkl-pipelines/pay-deploy/pact-broker.pkl
+   - ci/pkl-pipelines/common/**
+   uri: https://github.com/alphagov/pay-ci
+   username: alphagov-pay-ci-concourse
+ type: git

jobs:
  job set-pipeline has been removed:
- name: set-pipeline
- plan:
- - get: pact-broker-pipeline
-   trigger: true
- - file: pact-broker-pipeline/ci/pipelines/pact-broker.yml
-   set_pipeline: pact-broker

  job build-and-push-pact-broker has changed:
  name: build-and-push-pact-broker
  on_failure:
    attempts: 10
    params:
      channel: '#govuk-pay-starling'
      icon_emoji: ':concourse:'
      silent: true
      text: ':red-circle: Failed to build and push pay-pact-broker image - <https://pay-cd.deploy.payments.service.gov.uk/teams/$BUILD_TEAM_NAME/pipelines/$BUILD_PIPELINE_NAME/jobs/$BUILD_JOB_NAME/builds/$BUILD_NAME|Concourse
        build #$BUILD_NAME>'
      username: pay-concourse
    put: slack-notification
  on_success:
    attempts: 10
    params:
      channel: '#govuk-pay-activity'
      icon_emoji: ':concourse:'
      silent: true
      text: ':green-circle: Built and pushed pay-pact-broker image - <https://pay-cd.deploy.payments.service.gov.uk/teams/$BUILD_TEAM_NAME/pipelines/$BUILD_PIPELINE_NAME/jobs/$BUILD_JOB_NAME/builds/$BUILD_NAME|Concourse
        build #$BUILD_NAME>'
      username: pay-concourse
    put: slack-notification
  plan:
  - in_parallel:
      steps:
      - get: pact-broker-src
        trigger: true
      - get: pay-ci
  - file: pay-ci/ci/tasks/generate-docker-config-file.yml
    params:
      EMAIL: ((docker-email))
      PASSWORD: ((docker-access-token))
      USERNAME: ((docker-username))
    task: generate-docker-creds-config
  - config:
      image_resource:
        name: ""
        source:
          repository: alpine
          tag: latest
        type: registry-image
      inputs:
      - name: pact-broker-src
      outputs:
      - name: tags
      platform: linux
      run:
        args:
        - -ec
-       - |
-         cat pact-broker-src/ci/docker/pact-broker/Dockerfile | grep -i FROM | head -n 1 | awk '{print $2;}' | cut -f 2 -d ":" | tee tags/tag
+       - cat pact-broker-src/ci/docker/pact-broker/Dockerfile | grep -i FROM | head
+         -n 1 | awk '{print $2;}' | cut -f 2 -d ":" | tee tags/tag
        path: ash
    task: get-pact-broker-image-tag
  - config:
      image_resource:
        name: ""
        source:
          repository: concourse/oci-build-task
        type: registry-image
      inputs:
      - name: pact-broker-src
      - name: docker_creds
      outputs:
      - name: image
      platform: linux
      run:
        path: build
    params:
      CONTEXT: pact-broker-src/ci/docker/pact-broker
      DOCKER_CONFIG: docker_creds
      UNPACK_ROOTFS: "true"
    privileged: true
    task: build-pact-broker-image
  - get_params:
      skip_download: true
    params:
      additional_tags: tags/tag
      image: image/image.tar
    put: pact-broker-ecr-registry-deploy

  job update-pipeline has been added:
+ name: update-pipeline
+ plan:
+ - get: pipeline-source
+   trigger: true
+ - config:
+     image_resource:
+       name: ""
+       source:
+         repository: governmentdigitalservice/pay-concourse-runner
+         tag: latest
+       type: registry-image
+     inputs:
+     - name: pipeline-source
+     outputs:
+     - name: pipeline-source
+     platform: linux
+     run:
+       args:
+       - -c
+       - pkl eval 'pay-deploy/pact-broker.pkl' > 'pay-deploy/pact-broker.yaml'
+       dir: pipeline-source/ci/pkl-pipelines/
+       path: ash
+   task: generate-yaml-from-pkl
+ - file: pipeline-source/ci/pkl-pipelines/pay-deploy/pact-broker.yaml
+   set_pipeline: pact-broker

pipeline name: pact-broker
```

### CODE REVIEW

**Review by jfharden — CHANGES_REQUESTED** [review]

**Comment by alphagov-pay-ci-concourse** [comment]
**No YAML differences detected between PR and master pkl files**

**Comment by alphagov-pay-ci-concourse** [comment]
The following files had no diff in the YAML between the main branch and this PR:

```
pay-deploy/detect-secrets.yml
pay-deploy/pay-js.yml
pay-deploy/pkl-pipeline-changes.yml
pay-deploy/prometheus-pushgateway.yml
```

File: ci/pkl-pipelines/pay-deploy/pact-broker.pkl [hunk 1]
```diff
@@ -0,0 +1,170 @@
+amends "package://pkg.pkl-lang.org/github.com/alphagov/pkl-concourse-pipeline/pkl-concourse-pipeline@0.0.1#/Pipeline.pkl"
```
[the remark this answers is not in the document]
Comment (kbottla) [reply ci/pkl-pipelines/pay-deploy/pact-broker.pkl:1]: Updated! IntelliJ keeps removing the space though if I optimise imports 🤔

File: ci/pkl-pipelines/pay-deploy/pact-broker.pkl [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+          inputs = new Listing {
+            new TaskConfig.Input { name = "pact-broker-src" }
+          }
+          outputs = new Listing {
+            new TaskConfig.Output { name = "tags" }
+          }
+          run {
+            path = "ash"
+            args {
+                """
```
[the remark this answers is not in the document]
Comment (kbottla) [reply ci/pkl-pipelines/pay-deploy/pact-broker.pkl]: Updated to list of args.

**Comment by jfharden** [comment]
> **No YAML differences detected between PR and master pkl files**

Hmm I need to update this so it says "for pay-dev concourse team" or similar

**Review by jfharden — APPROVED** [review]

**Pull request merged**